### PR TITLE
[core] Minor improvement for worker pool, and add comments for protobuf

### PR DESCRIPTION
**Pull request by dentiny**
Some no-op change, including minor optimization for C++ coding, and add comments for something I find confused to understand.

### CODE REVIEW

File: src/ray/raylet/runtime_env_agent_client.cc [hunk 1]
```diff
@@ -235,7 +235,7 @@ class SessionPool {
         this->remove_session_from_running(session);
       });
     } else {
-      pending_sessions_.push(session);
```
Comment (dentiny) [inline src/ray/raylet/runtime_env_agent_client.cc:238]: `emplace` perform in-place construction.

File: src/ray/raylet/runtime_env_agent_client.cc [hunk 2]
```diff
@@ -244,10 +244,10 @@ class SessionPool {
   // enqueue one.
   void remove_session_from_running(std::shared_ptr<Session> session) {
     running_sessions_.erase(session);
-    if (pending_sessions_.size() > 0) {
```
Comment (dentiny) [inline src/ray/raylet/runtime_env_agent_client.cc:247]: Don't use `size` to decide empty to not, not all container guarantees O(1) `size` call.

**Review by jjyao — APPROVED** [review]

File: src/ray/raylet/runtime_env_agent_client.cc (hunk 2 above)
Comment (jjyao) [reply src/ray/raylet/runtime_env_agent_client.cc:247]: Is `empty()` guaranteed to be O(1)?

Comment (dentiny) [reply src/ray/raylet/runtime_env_agent_client.cc:247]: For all the containers that I know, yes.

Comment (dentiny) [reply src/ray/raylet/runtime_env_agent_client.cc:247]: A infamous example for O(N) `size` call is `std::list`:
- before C++11, there's no requirement from standard: https://en.cppreference.com/w/cpp/container/list/size
- quite a few discussions and debates: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=49561

**Comment by dentiny** [comment]
Hi @jjyao could you please help me merge this PR? Thank you!

**Pull request merged**